### PR TITLE
add data for day, week and month

### PR DESCRIPTION
**Pull request by wuqiank**
@rbran100
add daily, weekly and monthly data of sites and pages

### CODE REVIEW

File: ossap_stats/ossap_stats.module [hunk 1]
```diff
@@ -142,11 +142,11 @@ function ossap_stats_load_includes(){
 * ossap_stats_visited_sites()
 * no parameters. returns most frequently visited sites
 */
-function ossap_stats_visited_sites($output_json = FALSE){
+function ossap_stats_visited_sites($output_json = FALSE, $interval = 'day'){
   
   if($output_json) {
     //return output as a json object;
-    drupal_json_output(variable_get('visited_sites',array()));
+    drupal_json_output(variable_get('visited_sites_in_day',array()));
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module:149]: This variable should change depending on the $interval passed

File: ossap_stats/ossap_stats.module [hunk 2]
```diff
@@ -158,8 +158,8 @@ function ossap_stats_visited_sites($output_json = FALSE){
                    'hwpi.harvard.edu','projects.iq.harvard.edu', 'scholar.harvard.edu',
 	           'staging.hwpi.harvard.edu','staging.projects.iq.harvard.edu','staging.scholar.harvard.edu');
  //get results from the design_output function by sending the parameter of type = visits
-  
-  $results = ossap_stats_design_output('visits'); 
+  $type = 'sites';
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module]: What is the reason for changing the default to "sites" from "visits"?

File: ossap_stats/ossap_stats.module [hunk 3]
```diff
@@ -182,11 +182,11 @@ function ossap_stats_visited_sites($output_json = FALSE){
 * ossap_stats_viewed_pages()
 * no parameters. returns most frequently viewed pages
 */
-function ossap_stats_viewed_pages($output_json = FALSE){
+function ossap_stats_viewed_pages($output_json = FALSE, $interval = 'day'){
   
   //return output as a json object from the variable.
   if($output_json) {
-    drupal_json_output(variable_get('viewed_pages', array()));
+    drupal_json_output(variable_get('viewed_pages_in_day', array()));
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module]: This variable should also change depending on the interval or be an array.

File: ossap_stats/ossap_stats.module [hunk 4]
```diff
@@ -194,7 +194,8 @@ function ossap_stats_viewed_pages($output_json = FALSE){
   ossap_stats_load_includes();
   
   //get results from the design_output function by sending the parameter of type = pageviews
-  $results = ossap_stats_design_output('pageviews');
+  $type = 'pages';
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module]: Why the change from 'pageviews' to 'pages'?

File: ossap_stats/ossap_stats.module [hunk 5]
```diff
@@ -421,11 +458,23 @@ function ossap_stats_block_worker($stat = '') {
...
+    case 'visited_sites_in_week':
+      ossap_stats_most_visited_sites_in_week_cron_worker();
+      break;
+    case 'visited_sites_in_month':
+      ossap_stats_most_visited_sites_in_month_cron_worker();
       break;
-    case 'viewed_pages':
-      ossap_stats_most_viewed_pages_cron_worker();
+    case 'viewed_pages_in_day':
+      ossap_stats_most_viewed_pages_in_day_cron_worker();
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module]: We can use the same cron worker with arguments for day/week/month

File: ossap_stats/ossap_stats_credentials.inc [hunk 6]
```diff
@@ -71,12 +71,23 @@ function ossap_stats_design_output($type){
...
+  
+  // set time interval
+  if ($interval == 'day') {
+    $start_date = date('Y-m-d', strtotime('- 1 day'));
+  }
+  else if ($interval == 'week') {
+    $start_date = date('Y-m-d', strtotime('- 7 day'));
+  }
+  else if ($interval == 'month') {
+    $start_date = date('Y-m-d', strtotime('- 30 day'));
```
Comment (rbran100) [inline ossap_stats/ossap_stats_credentials.inc]: Super!

File: ossap_stats/ossap_stats.module (hunk 2 above)
Comment (wuqiank) [inline ossap_stats/ossap_stats.module]: Because previously in our .inc file, we used " if ($type == 'visits') {} " to indicate websites data and "else {}" to indicate pageviews, I think it is a little bit confusing, so i use 'sites' for sites and 'pages' for pages.

File: ossap_stats/ossap_stats.module (hunk 1 above)
Comment (wuqiank) [inline ossap_stats/ossap_stats.module:149]: yeah, will change this

File: ossap_stats/ossap_stats.module (hunk 5 above)
Comment (wuqiank) [inline ossap_stats/ossap_stats.module]: good point! Will do
